tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -0,0 +1,317 @@
+  using package_temporary_tags =
+      typename BoundaryCorrection::dg_package_data_temporary_tags;
+  using package_primitive_tags = detail::get_correction_primitive_vars<
+      System::has_primitive_and_conservative_vars, BoundaryCorrection>;
+
+  // Check that the temporary tags needed on the boundary are listed as
+  // temporary tags for the volume time derivative computation.
+  static_assert(
+      std::is_same_v<
+          tmpl::list_difference<package_temporary_tags, temporary_tags>,
+          tmpl::list<>>,
+      "There are temporary tags needed by the boundary correction that are not "
+      "computed as temporary tags by the system");
+
+  // Check that the primitive tags needed on the boundary are listed as the
+  // primiteve tags in the system.
+  static_assert(
+      std::is_same_v<tmpl::list_difference<package_primitive_tags,
+                                           primitive_variables_tags>,
+                     tmpl::list<>>,
+      "There are primitive tags needed by the boundary correction that are not "
+      "listed in the system as being primitive variables");
+
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<> dist(0.0, 1.0);
+  std::uniform_real_distribution<> pos_neg_dist(-1.0, 1.0);
+  DataVector used_for_size{face_mesh.number_of_grid_points()};
+
+  boost::optional<tnsr::I<DataVector, FaceDim + 1, Frame::Inertial>>
+      mesh_velocity{};
+  if (use_moving_mesh) {
+    mesh_velocity = make_with_random_values<
+        tnsr::I<DataVector, FaceDim + 1, Frame::Inertial>>(
+        make_not_null(&gen), make_not_null(&dist), used_for_size);
+  }
+
+  Variables<dg_package_field_tags> interior_package_data{used_for_size.size()};
+  const auto interior_fields_on_face = make_with_random_values<
+      Variables<tmpl::append<variables_tags, flux_tags, package_temporary_tags,
+                             package_primitive_tags>>>(
+      make_not_null(&gen), make_not_null(&dist), used_for_size);
+
+  Variables<dg_package_field_tags> exterior_package_data{used_for_size.size()};
+  const auto exterior_fields_on_face = make_with_random_values<
+      Variables<tmpl::append<variables_tags, flux_tags, package_temporary_tags,
+                             package_primitive_tags>>>(
+      make_not_null(&gen), make_not_null(&dist), used_for_size);
+
+  // Compute the interior and exterior normal vectors so they are pointing in
+  // opposite directions.
+  auto interior_unit_normal_covector = make_with_random_values<
+      tnsr::i<DataVector, FaceDim + 1, Frame::Inertial>>(
+      make_not_null(&gen), make_not_null(&pos_neg_dist), used_for_size);
+  const Scalar<DataVector> interior_normal_magnitude =
+      magnitude(interior_unit_normal_covector);
+  for (auto& t : interior_unit_normal_covector) {
+    t /= get(interior_normal_magnitude);
+  }
+  auto exterior_unit_normal_covector = interior_unit_normal_covector;
+  for (auto& t : exterior_unit_normal_covector) {
+    t *= -1.0;
+  }
```

Comment:
Design question: would it be useful to check the case where the normals are not simply the sign-flip of each other? I'm thinking of evolving GH and having (small) jumps in the normal normalization due to small jumps in the metric.

Reply:
at this stage there is no support for curved backgrounds. I already change the normal vectors slightly (by changing the spatial metric) in the curved spacetime case. That's just a few PRs out.